packages/teleterm/src/ui/services/workspacesService/documentsService/documentsUtils.ts

```diff
@@ -0,0 +1,25 @@
+import { routing } from 'teleterm/ui/uri';
+import { Document } from './types';
+
+export function getResourceUri(document: Document): string {
+  switch (document.kind) {
+    case 'doc.cluster':
+      return document.clusterUri;
+    case 'doc.gateway':
+      return document.targetUri;
+    case 'doc.terminal_tsh_node':
+      return document.serverUri;
+    case 'doc.terminal_tsh_kube':
+      return document.kubeUri;
+    case 'doc.terminal_shell':
+      return routing.getClusterUri(document);
```

Comment:
I'm considering adding a `clusterUri` field to `DocumentPtySession` (that field would be constructed in the same way as here, basing on `rootClusterId` and `leafClusterId`).

Reply:
Good idea, although I'd probably leave this for later. I looked into that a couple of weeks ago and that whole document could use some refactoring if we were to start changing those fields.